pkg/daemon/ceph/client/pool.go

```diff
@@ -326,11 +326,11 @@ func setCommonPoolProperties(context *clusterd.Context, clusterInfo *ClusterInfo
 		pool.Parameters = make(map[string]string)
 	}
 
-	if pool.Replicated.IsTargetRatioEnabled() {
+	if _, ok := pool.Parameters[targetSizeRatioProperty]; !ok {
 		pool.Parameters[targetSizeRatioProperty] = strconv.FormatFloat(pool.Replicated.TargetSizeRatio, 'f', -1, 32)
 	}
 
-	if pool.IsCompressionEnabled() {
+	if _, ok := pool.Parameters[CompressionModeProperty]; !ok {
```

Comment:
I see how the new approach was causing the empty string to be set, and causing the error. How about if we avoid setting it in the case of empty string? I believe this will fix it. 
```suggestion
	if _, ok := pool.Parameters[CompressionModeProperty]; !ok && pool.CompressionMode != "" {
```

Reply:
@travisn When I delete the CompressionMode parameter from the CephBlockPool CR, the compression_mode remains in passive mode.

```
2025-06-10 13:59:01.578207 I | ceph-spec: resource "CephBlockPool": "rook-ceph/replicapool3" spec has changed. diff=  v1.NamedBlockPoolSpec{
  	Name: "",
  	PoolSpec: v1.PoolSpec{
  		... // 2 identical fields
  		DeviceClass:        "",
  		EnableCrushUpdates: false,
- 		CompressionMode:    "passive",
+ 		CompressionMode:    "",
  		Replicated: v1.ReplicatedSpec{
  			Size:                     1,
- 			TargetSizeRatio:          0.2,
+ 			TargetSizeRatio:          0,
  			RequireSafeReplicaSize:   false,
  			ReplicasPerFailureDomain: 0,
  			... // 2 identical fields
  		},
  		ErasureCoded: {},
  		Parameters:   nil,
  		... // 5 identical fields
  	},
  }
2025-06-10 13:59:01.584516 I | ceph-spec: parsing mon endpoints: a=10.107.171.157:6789
2025-06-10 13:59:01.830859 I | ceph-block-pool-controller: creating pool "replicapool3" in namespace "rook-ceph"
2025-06-10 13:59:02.119647 I | clusterdisruption-controller: OSDs are up and PGs are clean. PG status: "all PGs in cluster are clean"
2025-06-10 13:59:02.119689 I | clusterdisruption-controller: `maxUnavailable` for the main OSD PDB is set to 1
2025-06-10 13:59:02.292948 I | cephclient: setting pool property "target_size_ratio" to "0" on pool "replicapool3"
2025-06-10 13:59:02.347639 I | clusterdisruption-controller: successfully reconciled OSD PDB controller
2025-06-10 13:59:03.527844 I | cephclient: application "rbd" is already set on pool "replicapool3"
2025-06-10 13:59:03.527888 I | cephclient: reconciling replicated pool replicapool3 succeeded
2025-06-10 13:59:03.527904 I | ceph-block-pool-controller: initializing pool "replicapool3" for RBD use
2025-06-10 13:59:04.330991 I | ceph-block-pool-controller: successfully initialized pool "replicapool3" for RBD use
2025-06-10 13:59:04.533553 I | ceph-block-pool-controller: enableStatsForPools="" 
2025-06-10 13:59:04.533565 I | op-config: deleting "mgr" "mgr/prometheus/rbd_stats_pools" option from the mon configuration database
2025-06-10 13:59:04.736322 I | op-config: successfully deleted "mgr/prometheus/rbd_stats_pools" option from the mon configuration database
2025-06-10 13:59:05.239478 I | clusterdisruption-controller: OSDs are up and PGs are clean. PG status: "all PGs in cluster are clean"
```

```
bash-5.1$ ceph osd pool get replicapool3 compression_mode
compression_mode: passive
bash-5.1$ 

```